CONTRIBUTING.md

```diff
@@ -1,35 +1,211 @@
+
+
+-   Specify which version of qBittorrent you're using.
+-   Specify the name and version of the OS you're using.
+-   Provide a step-by-step description of the suggested enhancement in as many details as possible.
+-   Describe the current behavior and explain which behavior you expected to see instead and why.
+-   Include screenshots and animated GIFs which help you demonstrate the steps or point out the part of qBittorrent which the suggestion is related to.
+-   If this enhancement exists in other bittorrent clients, list list those clients.
 
 
 # Opening a pull request
 
 ### Must read
-* Read our [**coding guidelines**](https://github.com/qbittorrent/qBittorrent/blob/master/CODING_GUIDELINES.md). There are some scripts to help you: [uncrustify script](https://raw.githubusercontent.com/qbittorrent/qBittorrent/master/uncrustify.cfg), [astyle script](https://gist.github.com/Chocobo1/539cee860d1eef0acfa6), [(related thread)](https://github.com/qbittorrent/qBittorrent/issues/2192).
-* Keep the title **short** and provide a **clear** description about what your pull request does.
-* Provide **screenshots** for UI related changes.
-* Keep your git commit history **clean** and **precise**. Commits like `xxx fixup` should not appear.
-* If your commit fix a reported issue (for example #4134), add the following message to the commit `Closes #4134.`. Example [here](https://github.com/qbittorrent/qBittorrent/commit/a74bac20c4e8de9776bf9bb77fdc7526135d1988).
+*   Read our [**coding guidelines**](https://github.com/qbittorrent/qBittorrent/blob/master/CODING_GUIDELINES.md). There are some scripts to help you: [uncrustify script](https://raw.githubusercontent.com/qbittorrent/qBittorrent/master/uncrustify.cfg), [astyle script](https://gist.github.com/Chocobo1/539cee860d1eef0acfa6), [(related thread)](https://github.com/qbittorrent/qBittorrent/issues/2192).
+*   Keep the title **short** and provide a **clear** description about what your pull request does.
+*   Provide **screenshots** for UI related changes.
+*   Keep your git commit history **clean** and **precise**. Commits like `xxx fixup` should not appear.
+*   If your commit fix a reported issue (for example #4134), add the following message to the commit `Closes #4134.`. Example [here](https://github.com/qbittorrent/qBittorrent/commit/a74bac20c4e8de9776bf9bb77fdc7526135d1988).
 
 ### Good to know
-* **Search** pull request history! Others might already implemented your idea and is waiting to be merged (or got rejected already). Save your precious time by doing a search first.
-* When resolving merge conflicts, do `git rebase <target_branch_name>`, don't do `git pull`. Then you can start fixing the conflicts. Here is a good explanation: [link](https://www.atlassian.com/git/tutorials/merging-vs-rebasing).
+*   **Search** pull request history! Others might already implemented your idea and is waiting to be merged (or got rejected already). Save your precious time by doing a search first.
```

Comment:
Others might _have_ already implemented your idea and _it_ is waiting to be merged (or _it_ got rejected already).

Reply:
Ok